### ISSUE TITLE
Crashes when launching a Gtk4 application

### ISSUE THREAD

**Issue by jbeich** [djpohly/dwl#173]
### Steps to reproduce
1. Start dwl
2. Start gtk4-demo (via key or from a terminal)

### Details
```c
* thread #1, name = 'dwl', stop reason = signal SIGSEGV: invalid address (fault address: 0x48)
    frame #0: 0x000000000020b8fb dwl`setfullscreen(c=0x000000083c738cc0, fullscreen=1) at dwl.c:1036:23
   1033                 c->prevy = c->geom.y;
   1034                 c->prevheight = c->geom.height;
   1035                 c->prevwidth = c->geom.width;
-> 1036                 resize(c, c->mon->m.x, c->mon->m.y, c->mon->m.width, c->mon->m.height, 0);
   1037         } else {
   1038                 /* restore previous size instead of arrange for floating windows since
   1039                  * client positions are set by the user and cannot be recalculated */
(lldb) bt
* thread #1, name = 'dwl', stop reason = signal SIGSEGV: invalid address (fault address: 0x48)
  * frame #0: 0x000000000020b8fb dwl`setfullscreen(c=0x000000083c738cc0, fullscreen=1) at dwl.c:1036:23
    frame #1: 0x000000000020b5fc dwl`fullscreennotify(listener=0x000000083c738d70, data=0x00007fffffffda60) at dwl.c:1049:2
    frame #2: 0x0000000801ee71ba libwlroots.so.10`wlr_signal_emit_safe(signal=0x000000083c73d8d0, data=0x00007fffffffda60) at signal.c:29:3
    frame #3: 0x0000000801eb73ba libwlroots.so.10`xdg_toplevel_handle_unset_fullscreen(client=0x000000081d22b700, resource=0x000000081d113e80) at wlr_xdg_toplevel.c:404:2
    frame #4: 0x0000000811bfb0dd libffi.so.7`ffi_call_unix64 at unix64.S:101
    frame #5: 0x0000000811bfa3b2 libffi.so.7`ffi_call_int(cif=0x00007fffffffdbd0, fn=(libwlroots.so.10`xdg_toplevel_handle_unset_fullscreen at wlr_xdg_toplevel.c:392), rvalue=0x0000000000000000, avalue=0x00007fffffffdc00, closure=0x0000000000000000) at ffi64.c:669:3
    frame #6: 0x0000000811bf9f46 libffi.so.7`ffi_call(cif=0x00007fffffffdbd0, fn=(libwlroots.so.10`xdg_toplevel_handle_unset_fullscreen at wlr_xdg_toplevel.c:392), rvalue=0x0000000000000000, avalue=0x00007fffffffdc00) at ffi64.c:688:3
    frame #7: 0x00000008027f1628 libwayland-server.so.0`wl_closure_invoke(closure=0x000000083c6f33a0, flags=2, target=0x000000081d113e80, opcode=12, data=0x000000081d22b700) at connection.c:1025:2
    frame #8: 0x00000008027e943d libwayland-server.so.0`wl_client_connection_data(fd=30, mask=1, data=0x000000081d22b700) at wayland-server.c:437:4
    frame #9: 0x00000008027ed467 libwayland-server.so.0`wl_event_source_fd_dispatch(source=0x000000081d11d180, ep=0x00007fffffffde80) at event-loop.c:112:9
    frame #10: 0x00000008027eeb54 libwayland-server.so.0`wl_event_loop_dispatch(loop=0x000000081d0b0320, timeout=-1) at event-loop.c:1027:4
    frame #11: 0x00000008027ea8ef libwayland-server.so.0`wl_display_run(display=0x000000081d0da000) at wayland-server.c:1408:3
    frame #12: 0x0000000000207dd5 dwl`run(startup_cmd=0x0000000000000000) at dwl.c:1861:2
    frame #13: 0x00000000002075f9 dwl`main(argc=1, argv=0x00007fffffffe108) at dwl.c:2609:2
    frame #14: 0x0000000000207330 dwl`_start(ap=<unavailable>, cleanup=<unavailable>) at crt1_c.c:73:7
(lldb) p *c
(Client) $0 = {
  link = {
    prev = NULL
    next = NULL
  }
  flink = {
    prev = NULL
    next = NULL
  }
  slink = {
    prev = NULL
    next = NULL
  }
  surface = {
    xdg = 0x000000083c738b80
    xwayland = 0x000000083c738b80
  }
  commit = {
    link = {
      prev = 0x000000083c738c30
      next = 0x000000081d12b230
    }
    notify = 0x000000000020b300 (dwl`commitnotify at dwl.c:776)
  }
  map = {
    link = {
      prev = 0x000000083c738c78
      next = 0x000000083c738c78
    }
    notify = 0x000000000020b360 (dwl`mapnotify at dwl.c:1299)
  }
  unmap = {
    link = {
      prev = 0x000000083c738c88
      next = 0x000000083c738c88
    }
    notify = 0x000000000020b460 (dwl`unmapnotify at dwl.c:2283)
  }
  destroy = {
    link = {
      prev = 0x000000083c738c48
      next = 0x000000083c738c48
    }
    notify = 0x000000000020b4d0 (dwl`destroynotify at dwl.c:999)
  }
  set_title = {
    link = {
      prev = 0x000000083c73d930
      next = 0x000000083c73d930
    }
    notify = 0x000000000020b560 (dwl`updatetitle at dwl.c:2334)
  }
  fullscreen = {
    link = {
      prev = 0x000000083c73d8d0
      next = 0x00007fffffffda10
    }
    notify = 0x000000000020b5c0 (dwl`fullscreennotify at dwl.c:1047)
  }
  geom = (x = 0, y = 0, width = 0, height = 0)
  mon = NULL
  bw = 0
  tags = 0
  isfloating = 0
  isurgent = 0
  resize = 0
  prevx = 0
  prevy = 0
  prevwidth = 0
  prevheight = 0
  isfullscreen = 1
}
```
### Environment
```c
$ git describe --tags --always
v0.2.2

$ cc --version
FreeBSD clang version 13.0.0 (git@github.com:llvm/llvm-project.git llvmorg-13.0.0-0-gd7b669b3a303)
Target: x86_64-unknown-freebsd14.0
Thread model: posix
InstalledDir: /usr/bin

$ pkg info -x wlroots wayland libxcb xkbcommon libinput
wlroots-0.15.0
wayland-1.20.0
wayland-protocols-1.24
xwayland-devel-21.0.99.1.156
libxcb-1.14_1
libxkbcommon-1.3.1
libinput-1.19.1_1
```

**Comment by raphaelr** [comment]
I think this is a duplicate of #170

**Comment by jbeich** [comment]
With #171 the crash is gone but Gtk4 windows overlap others e.g.,

## dwl
![dwl](https://user-images.githubusercontent.com/11153579/147244922-8e8fc18a-efac-4af8-b0b2-5c29e30a9765.png)
## river
![river](https://user-images.githubusercontent.com/11153579/147244946-cf0bc30f-27fa-454b-ae0c-97c207505d00.png)
## sway
![sway](https://user-images.githubusercontent.com/11153579/147244964-bf2d637c-a2c4-4306-94a3-4a2fdf8256c0.png)

**Comment by Sevz17** [comment]
Fixed in 88f241d1cf54ffb7ec1f6e092ccb547c96f57ac9

**Comment by Sevz17** [comment]
The windows overlapping continues, but at least it doesn't crash :)
